rad/src/error.rs

```diff
@@ -117,6 +117,12 @@ pub enum RadError {
     /// Overflow error
     #[fail(display = "Overflow error")]
     Overflow,
+    /// Mismatching types
+    #[fail(display = "Mismatching types")]
+    MismatchingTypes,
+    /// Arrays to be reduced have different sizes
+    #[fail(display = "Arrays to be reduced have different sizes")]
```

Comment:
- For context, e should explicitly state which reducer thrown the error.
- We can print the sizes here as `(x != y)`